test/unit_tests/threads/test_multicore.cpp

```diff
@@ -0,0 +1,248 @@
+#include <catch2/catch.hpp>
+
+#include <hocdec.h>
+#include <ocfunc.h>
+#include <code.h>
+#include <thread>
+#include <algorithm>
+#include <vector>
+#include "../unit_test.h"
+
+#ifdef NRN_ENABLE_THREADS
+#include <multicore.h>
+#include <nrn_ansi.h>
+
+extern int use_cachevec;
+extern int nrn_how_many_processors();
+
+/* @brief
+ *  Test multicore implementation:
+ *  * parallel mode (std::threads)
+ *  * parallel mode with busywait
+ *  * serial mode
+ *  * performance
+ *      * NOTE: GitHub runners don't have enough capabilities for performance KPIs
+ */
+
+
+// generate a range starting from 1 and doubling until we reach the nof concurrent threads
+auto make_available_threads_range() {
+    auto nof_threads_range =
+        std::vector<int>(1 + static_cast<int>(std::log2(nrn_how_many_processors())), 1);
+    std::generate(nof_threads_range.begin() + 1, nof_threads_range.end(), [n = 1]() mutable {
+        return n *= 2;
+    });
+    return nof_threads_range;
+}
+TEST_CASE("Multicore unit and performance testing", "[NEURON][multicore]") {
+    SECTION("Simulation set-up", "[NEURON][multicore][setup]") {
+        GIVEN("We create a parallel context") {
+            REQUIRE(hoc_oc("objref pc\n"
+                           "pc = new ParallelContext()") == 0);
+            THEN("we assert nrn_thread is equal to 1") {
+                REQUIRE(nrn_nthread == 1);
+            }
+            THEN("we check we have no worker threads") {
+                REQUIRE(nof_worker_threads() == 0);
+            }
+            THEN("we setup 1000 passive membrane cells for the simulation") {
+                REQUIRE(hoc_oc(pass_cell_template) == 0);
+                REQUIRE(hoc_oc(prun) == 0);
+                std::string cells{1000_pas_cells};
```

Comment:
I'm not sure what I think of the `1000_pas_cells` syntax 🙈

Reply:
`std::string pas_cell_hoc_code(std::string num_cells)`?

Reply:
I just think it reads better as-is, don't have a strong opinion

Reply:
Not a big fan of the extra where-is-that-variable-defined-oh-wait-it's-not-a-variable cognitive journey :)